modules/table/src/integrationTest/java/org/apache/ignite/internal/table/distributed/disaster/ItHighAvailablePartitionsRecoveryTest.java

```diff
@@ -297,6 +301,56 @@ void testHaZoneScaleDownNodesDoNotRemovedFromStable() throws InterruptedExceptio
         waitAndAssertStableAssignmentsOfPartitionEqualTo(node, HA_TABLE_NAME, PARTITION_IDS, allNodes);
     }
 
+    @Test
+    void testRebalanceInHaZone() throws InterruptedException {
+        createHaZoneWithTable();
+
+        startNode(3);
+
+        Set<String> fourNodes = runningNodes().map(Ignite::name).collect(Collectors.toUnmodifiableSet());
+
+        executeSql(format("ALTER ZONE %s SET REPLICAS=%d", HA_ZONE_NAME, 4));
+
+        IgniteImpl node = igniteImpl(0);
+
+        waitAndAssertStableAssignmentsOfPartitionEqualTo(node, HA_TABLE_NAME, PARTITION_IDS, fourNodes);
+
+        stopNode(3);
+
+        Set<String> threeNodes = runningNodes().map(Ignite::name).collect(Collectors.toUnmodifiableSet());
+
+        executeSql(format("ALTER ZONE %s SET data_nodes_auto_adjust_scale_down=%d", HA_ZONE_NAME, 1));
+
+        waitAndAssertStableAssignmentsOfPartitionEqualTo(node, HA_TABLE_NAME, PARTITION_IDS, threeNodes);
```

Comment:
let's add
`Verify that the data is consistent across all nodes (A, B, C, D). `
as it presented in the doc